cases/services/tag_enricher.py

```
@@ -0,0 +1,1349 @@
+            llm_invoked = True
+            try:
+                llm_tags = self._classify_with_llm(case)
+                if llm_tags:
+                    new_tags = list(dict.fromkeys(tags + llm_tags))
+                    llm_contributed = len(new_tags) > len(tags)
+                    all_tags = new_tags
+                    logger.info(f"  + metadata_llm succeeded: {len(all_tags)} tags")
+                else:
+                    all_tags = tags
+                    logger.info("  - metadata_llm returned no tags, using rule-based")
+            except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)